src/main/java/com/esri/samples/map/generate_offline_map_with_local_basemap/GenerateOfflineMapWithLocalBasemap.java

```diff
@@ -0,0 +1,331 @@
+
+  private MapView mapView;
+  private String downloadedBasemapSavedLocation;
+  private String localBasemapDirectory;
+  private File localBasemapFile;
+  private GraphicsOverlay graphicsOverlay;
+  private Graphic downloadArea;
+  private ArcGISMap map;
+
+  private ProgressBar progressBar;
+  private Button offlineMapButton;
+
+  private GenerateOfflineMapParameters generateOfflineMapParameters;
+  private GenerateOfflineMapJob generateOfflineMapJob;
+  private OfflineMapTask task;
+
+
+  @Override
+  public void start(Stage stage) throws Exception {
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+      scene.getStylesheets().add(getClass().getResource("/css/style.css").toExternalForm());
+
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Generate Offline Map With Local Basemap Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a map view
+      mapView = new MapView();
+
+      // create a button to take the map offline
+      createTakeMapOfflineButton();
+
+      // create a portal item with the itemId of the web map
+      Portal portal = new Portal("https://www.arcgis.com");
+      PortalItem portalItem = new PortalItem(portal, "acc027394bc84c2fb04d1ed317aac674");
+
+      // create a map with the portal item
+      map = new ArcGISMap(portalItem);
+      addMapLoadedListener();
+      // set the map to the map view
+      mapView.setMap(map);
+      // update the download area box whenever the viewpoint changes
+      mapView.addViewpointChangedListener(viewpointChangedEvent -> updateDownloadArea());
+
+      // create a progress bar to show download progress
+      createProgressIndicator();
+
+      // when the take map offline button is clicked, start the offline map task job
+      offlineMapButton.setOnAction(e -> {
+        try {
+          offlineMapButton.setVisible(false);
+          progressBar.setVisible(true);
+
+          // specify the extent, min scale, and max scale as parameters
+          double minScale = mapView.getMapScale();
+          double maxScale = map.getMaxScale();
+          // minScale must always be larger than maxScale
+          if (minScale <= maxScale) {
+            minScale = maxScale + 1;
+          }
+
+          // create an offline map task with the map
+          task = new OfflineMapTask(map);
+          // create default generate offline map parameters
+          ListenableFuture<GenerateOfflineMapParameters> generateOfflineMapParametersListenableFuture = task.createDefaultGenerateOfflineMapParametersAsync(downloadArea.getGeometry(), minScale, maxScale);
+          // get the offline map parameters from the offline map task
+          generateOfflineMapParameters = generateOfflineMapParametersListenableFuture.get();
```

Comment:
Move this into the done listener callback. Otherwise it will just block the thread until done. This will also move the try/catch inside the done listener. See other samples with listenable futures for reference.